soc/arm/aspeed/ast10x0/tools/gen_uart_booting_image.sh

```diff
@@ -0,0 +1,38 @@
+#!/bin/sh
```

Comment:
I would recommend rewriting this script in Python (v3), not because I'm personally against bash (or whatever shell happens to be installed on a given system), but because Python is a more consistent scripting environment that works across all platforms supported by Zephyr.

Is that doable?

Reply:
Hi @cfriedt ,

Yes, it is doable.  And just now I edit my first commit to replace the shell script by the Python based tool.  Please kindly review my change, thank you very much.